src/transactionIntentsFactories/smartContractTransactionIntentsFactory.spec.ts

```diff
@@ -0,0 +1,108 @@
+import { assert, expect } from "chai";
+import { SmartContractTransactionIntentsFactory } from "./smartContractTransactionIntentsFactory";
+import { Address } from "../address";
+import { Code } from "../smartcontracts/code";
+import { promises } from "fs";
+import { AbiRegistry } from "../smartcontracts/typesystem/abiRegistry";
+import { U32Value } from "../smartcontracts";
+import { CONTRACT_DEPLOY_ADDRESS } from "../constants";
+
+describe("test smart contract intents factory", function () {
+    let smartContractIntentsFactory: SmartContractTransactionIntentsFactory;
+    let abiSmartContractIntentsFactory: SmartContractTransactionIntentsFactory;
+    let adderByteCode: Code;
+    let abiRegistry: AbiRegistry;
+
+    before(async function () {
+        smartContractIntentsFactory = new SmartContractTransactionIntentsFactory({ chainID: "D", minGasLimit: 50000, gasLimitPerByte: 1500 });
```

Comment:
Rename to simply `factory` (not ambiguous in the context of this file).

Reply:
Renamed to `factory`